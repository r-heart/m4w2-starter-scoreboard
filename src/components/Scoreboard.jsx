import PropTypes from "prop-types";
import { formatMinutesAndSeconds } from "../utils";

export default function Scoreboard({
  homeScore,
  awayScore,
  time,
  currentPeriod,
}) {
  return (
    <section className="flex flex-row gap-x-8">
      <div className="flex flex-col items-center">
        <h2 className="text-2xl">Home</h2>
        <p className="text-4xl" data-testid="home-score">
          {homeScore}
        </p>
      </div>

      <div className="flex flex-col items-center gap-y-4">
        <div className="flex flex-col items-center">
          <h2 className="text-2xl">Period</h2>
          <p className="text-4xl" data-testid="period">
            {currentPeriod}
          </p>
        </div>

        <div className="flex flex-col items-center">
          <h2 className="text-2xl">Time</h2>
          <p className="text-4xl" data-testid="time">
            {formatMinutesAndSeconds(time)}
          </p>
        </div>
      </div>

      <div className="flex flex-col items-center">
        <h2 className="text-2xl">Away</h2>
        <p className="text-4xl" data-testid="away-score">
          {awayScore}
        </p>
      </div>
    </section>
  );
}

Scoreboard.propTypes = {
  homeScore: PropTypes.number,
  awayScore: PropTypes.number,
  time: PropTypes.number,
  currentPeriod: PropTypes.number,
};

Scoreboard.defaultProps = {
  homeScore: 0,
  awayScore: 0,
  time: 900,
  currentPeriod: 1,
};
