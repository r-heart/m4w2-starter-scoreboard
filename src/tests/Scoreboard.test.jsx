import { render } from "@testing-library/react";
import Scoreboard from "../components/Scoreboard";

test("renders the scoreboard", () => {
  const rendered = render(<Scoreboard />);
  expect(rendered).toMatchSnapshot();
});
