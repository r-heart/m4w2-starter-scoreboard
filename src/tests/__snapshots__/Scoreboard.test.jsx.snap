// Vitest Snapshot v1

exports[`renders the scoreboard 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="flex flex-row gap-x-8"
      >
        <div
          class="flex flex-col items-center"
        >
          <h2
            class="text-2xl"
          >
            Home
          </h2>
          <p
            class="text-4xl"
            data-testid="home-score"
          >
            0
          </p>
        </div>
        <div
          class="flex flex-col items-center gap-y-4"
        >
          <div
            class="flex flex-col items-center"
          >
            <h2
              class="text-2xl"
            >
              Period
            </h2>
            <p
              class="text-4xl"
              data-testid="period"
            >
              1
            </p>
          </div>
          <div
            class="flex flex-col items-center"
          >
            <h2
              class="text-2xl"
            >
              Time
            </h2>
            <p
              class="text-4xl"
              data-testid="time"
            >
              NaN:NaN
            </p>
          </div>
        </div>
        <div
          class="flex flex-col items-center"
        >
          <h2
            class="text-2xl"
          >
            Away
          </h2>
          <p
            class="text-4xl"
            data-testid="away-score"
          >
            0
          </p>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="flex flex-row gap-x-8"
    >
      <div
        class="flex flex-col items-center"
      >
        <h2
          class="text-2xl"
        >
          Home
        </h2>
        <p
          class="text-4xl"
          data-testid="home-score"
        >
          0
        </p>
      </div>
      <div
        class="flex flex-col items-center gap-y-4"
      >
        <div
          class="flex flex-col items-center"
        >
          <h2
            class="text-2xl"
          >
            Period
          </h2>
          <p
            class="text-4xl"
            data-testid="period"
          >
            1
          </p>
        </div>
        <div
          class="flex flex-col items-center"
        >
          <h2
            class="text-2xl"
          >
            Time
          </h2>
          <p
            class="text-4xl"
            data-testid="time"
          >
            NaN:NaN
          </p>
        </div>
      </div>
      <div
        class="flex flex-col items-center"
      >
        <h2
          class="text-2xl"
        >
          Away
        </h2>
        <p
          class="text-4xl"
          data-testid="away-score"
        >
          0
        </p>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
